src/CSnakes.Runtime/Python/PyObject.cs

```diff
@@ -279,12 +279,23 @@ public override string ToString()
         }
     }
 
-    public T As<T>() => (T)(td.ConvertTo(this, typeof(T)) ?? default!);
+    public T As<T>()
+    {
+        using (GIL.Acquire())
+        {
+            return (T)(td.ConvertTo(this, typeof(T)) ?? default!);

```

Comment:
Instead of `default!`, I would update this to be `T?`. It is annoying in almost all cases, but it is clearer than someone getting a `null` when it doesn't indicate that. Alternatively, one could throw if the conversation fails? The name of the function does seem to indicate that `T?` is likely more appropriate though.

Reply:
If the conversion fails, there is always an exception thrown so the response type isn't nullable anyway. We only have to do this because of the `TypeConverter` API. Will add to the list of things to fix when that is refactored.